src/doc/unstable-book/src/language-features/cmse-nonsecure-entry.md

```diff
@@ -0,0 +1,75 @@
+# `cmse_nonsecure_entry`
+
+The tracking issue for this feature is: [#75835]
+
+[#75835]: https://github.com/rust-lang/rust/issues/75835
+
+------------------------
+
+The [TrustZone-M
+feature](https://developer.arm.com/documentation/100690/latest/) is available
+for targets with the Armv8-M architecture profile (`thumbv8m` in their target
+name).
+LLVM, the Rust compiler and the linker are providing
+[support](https://developer.arm.com/documentation/ecm0359818/latest/) for the
+TrustZone-M feature.
+
+One of the thing provided, with this unstable feature, is the
+`cmse_nonsecure_entry` attribute.  This attribute marks a Secure function as an
+entry function (see [section
+5.4](https://developer.arm.com/documentation/ecm0359818/latest/) for details).
+With this attribute, the compiler will do the following:
+* add a special symbol on the function which is the `__acle_se_` prefix and the
+  standard function name
+* constrain the number of parameters to avoid using the Non-Secure stack
```

Comment:
I think I agree with that, an LLVM error will be a bit more cryptic for the user. I am going to look into how and were I can do that. Basically we want to emit an error if the function is going to pass any of its parameter using the stack as opposed to using only registers.

Reply:
I think it would also be fine to do this as follow-up work if you want. Just add it to the blockers in the tracking issue since it's important to do before stabilization.

Reply:
The LLVM check was accidentally omitted and [is being reviewed](https://reviews.llvm.org/D86478) now.
I am happy for this to be done as a follow-up work but I think the LLVM patch should be there (once merged). If not, entry functions declaring more than 32 bytes of arguments would at best provoke a segmentation fault but at worse expose secure data...

Or is that fine for unstable features not to be safe/secure if that is documented somewhere?

Reply:
Ah, right, yeah. If this can cause segfaults we should wait for that fix.

Reply:
Then I guess I can follow [this guide](https://rustc-dev-guide.rust-lang.org/backend/updating-llvm.html#bugfix-updates) to cherry-pick the bugfix into the LLVM fork Rust uses.

Reply:
Regarding tests, would it work if I add a UI test with a `// compile-flags: --target thumbv8m.main-none-eabi` option ([similar to this](https://github.com/rust-lang/rust/blob/master/src/test/ui/issues/issue-49851/compiler-builtins-error.rs)) and that I check that the LLVM error is generated when declaring an entry function with too many parameters?

Reply:
In addition to that, you'll need to let the test runner know that the test only works on the `thumbv8m.main-none-eabi` target so it is skipped when that target isn't being built. I think that can be done with this directive:

`// only-thumbv8m.main-none-eabi`

Could be wrong though.

Reply:
The LLVM bug fix is now merged! I made [a PR](https://github.com/rust-lang/llvm-project/pull/76) to cherry-pick it